app/routes.py

```diff
@@ -1,2 +1,286 @@
+                })
+            else:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": task.completed_at
+                })
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if request_body.get("title") and request_body.get("description"): 
+            if request_body.get("completed_at")==None:
+                new_task = Task(title=request_body["title"],
+                                description=request_body["description"])
+                db.session.add(new_task)
+                db.session.commit()
+                response_value = {"task":{
+                        "id": new_task.task_id,
+                        "title": new_task.title,
+                        "description": new_task.description,
+                        "is_complete": False
+                }}
+            elif request_body.get("completed_at"):
+                new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=datetime.now())
+                db.session.add(new_task)
+                db.session.commit()
+                response_value = {"task":{
+                        "id": new_task.task_id,
+                        "title": new_task.title,
+                        "description": new_task.description,
+                        "is_complete": True
+                    }}
+        else:
+            return make_response({"details":"Invalid data"}, 400)
+
+        return make_response(response_value, 201)
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        if not task.completed_at:
+            if task.goal_id:
+                response_value = {"task":{
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False
+                }}
+            else:
+                response_value = {"task":{
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False
+                }}
+
+            return make_response(response_value, 200)
+        else:
+            response_value = {"task":{
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at
+            }}
+
+            return make_response(response_value, 200)
```

Comment:
Notice how we're returning 3 versions of the same dictionary? To avoid rewriting the same dictionary over and over again, we can create a 'base' dictionary then add an additional key-value pair if a goal id exists for that task.

```suggestion

        task_dict = {
                    "id": task.task_id,
                    "title": task.title,
                    "description": task.description,
                    "is_complete": True if task.completed_at else False
                }
       
        if task.goal_id:
            task_dict["goal_id"] = task.goal_id
             
        response_value = { "task": task_dict }
        return make_response(response_value, 200)

```